File: 프로그래머스/lv1/118666. 성격 유형 검사하기/성격 유형 검사하기.py
def solution(survey, choices):
    scores = {"A":0, "N":0, "C":0, "F":0, "M":0, "J":0, "R":0, "T":0}
    for idx, choice in enumerate(choices):
        if choice - 4 > 0:
            scores[survey[idx][1]] += choice - 4
        elif choice - 4 < 0:
            scores[survey[idx][0]] += 4 - choice

    answer = ""
    if scores["R"] >= scores["T"]:
        answer += "R"
    else:
        answer += "T"

    if scores["C"] >= scores["F"]:
        answer += "C"
    else:
        answer += "F"

    if scores["J"] >= scores["M"]:
        answer += "J"
    else:
        answer += "M"

    if scores["A"] >= scores["N"]:
        answer += "A"
    else:
        answer += "N"

    return answer

File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.java
import java.util.*;

class Solution {
    static Map<String, Integer> map = new HashMap<>(){{
        put("R",0);
        put("T",0);

        put("C",0);
        put("F",0);

        put("J",0);
        put("M",0);

        put("A",0);
        put("N",0);
    }};
    public String solution(String[] survey, int[] choices) {
        for (int i = 0; i < choices.length; i++) {
            int choice = choices[i];
            String word = "";
            if(choice < 4) {
                word = String.valueOf(survey[i].charAt(0));
                map.put(word, map.getOrDefault(word, 0) + 4 - choice);
            }
            else{
                word = String.valueOf(survey[i].charAt(1));
                map.put(word, map.getOrDefault(word, 0) + choice - 4);
            }

        }
//        System.out.println(map);
        StringBuilder answer = new StringBuilder();
        answer
                .append(map.get("R") >= map.get("T") ? "R" : "T")
                .append(map.get("C") >= map.get("F") ? "C" : "F")
                .append(map.get("J") >= map.get("M") ? "J" : "M")
                .append(map.get("A") >= map.get("N") ? "A" : "N");

        System.out.println(answer);
        return answer.toString();
    }
}